src/simulation/m_lag_bubbles.fpp

```diff
@@ -0,0 +1,1746 @@
+
+            !Control volume radius
+            if (present(Romega)) Romega = dc
+
+            ! Getting p_inf
+            if (ptype == 1) then
+                f_pressure_inf = f_pressure_inf + preterm1*term1 + term2
+            end if
+
+        end if
+
+    end function f_pressure_inf
+
+    !>  This subroutine updates the Euler-Lagrange temporal variables before entering to the next time-stage in the time stepper.
+        !! @param RKstep Current time step in the adaptative stepper
+        !! @param q_cons_ts Conservative variables
+        !! @param rhs_ts Time derivatives of the conservative variables
+        !! @param q_prim_vf Primitive variables
+    subroutine s_update_tmp_rkck(RKstep, q_cons_ts, rhs_ts, q_prim_vf)
+
+        integer, intent(in) :: RKstep
+        type(vector_field), dimension(:), intent(inout) :: q_cons_ts
+        type(vector_field), dimension(:), intent(inout) :: rhs_ts
+        type(scalar_field), dimension(:), intent(inout) :: q_prim_vf
+
+        integer :: i, j, k, l, q, ierr
+        logical :: change, indomain
+        integer, dimension(3) :: cell
+        real(kind(0.d0)) radiusOld, velOld
+        logical :: aux
+
+        call s_transfer_data_to_tmp()
+
+        !$acc parallel loop gang vector default(present) reduction(IOR: lag_largestep) private(k) copyin(RKstep)
+        do k = 1, nBubs
+
+            radiusOld = lag_bub_interface_rad(k, 2)
+            velOld = lag_bub_interface_vel(k, 2)
+
+            !$acc loop seq
+            do i = 1, RKstep
+                lag_bub_interface_rad(k, 2) = lag_bub_interface_rad(k, 2) + dt*lag_RKcoef(RKstep, i)*lag_bub_interface_draddt(k, i)
+                lag_bub_interface_vel(k, 2) = lag_bub_interface_vel(k, 2) + dt*lag_RKcoef(RKstep, i)*lag_bub_interface_dveldt(k, i)
+                lag_bub_motion_pos(k, 1:3, 2) = lag_bub_motion_pos(k, 1:3, 2) + dt*lag_RKcoef(RKstep, i)*lag_bub_motion_dposdt(k, 1:3, i)
+                lag_bub_motion_vel(k, 1:3, 2) = lag_bub_motion_vel(k, 1:3, 2) + dt*lag_RKcoef(RKstep, i)*lag_bub_motion_dveldt(k, 1:3, i)
+                lag_bub_gas_p(k, 2) = lag_bub_gas_p(k, 2) + dt*lag_RKcoef(RKstep, i)*lag_bub_gas_dpdt(k, i)
+                lag_bub_gas_mv(k, 2) = lag_bub_gas_mv(k, 2) + dt*lag_RKcoef(RKstep, i)*lag_bub_gas_dmvdt(k, i)
+            end do
+
+            if ((lag_bub_interface_rad(k, 2) <= 0.0d0) .or. &                    ! no negative radius
+                (lag_bub_motion_pos(k, 1, 2) /= lag_bub_motion_pos(k, 1, 2))) then   ! finite bubble location
+
+                print *, 'Large time step. Radius from:', radiusOld, ' to :', lag_bub_interface_rad(k, 2), &
+                    '; and velocity from', velOld, ' to :', lag_bub_interface_vel(k, 2)
+                lag_largestep = .true.
+                if (dt < 2.d-15) then
+                    print *, 'WARNING large step', dt, '. Removing bubble:', lag_bub_id(k, 1)
+                    call s_remove_lag_bubble(k)
+                end if
+                goto 711
```

Comment:
Can't do this on GPUs as the outer loop (k) isn't sequential

Reply:
Thanks @anandrdbz. I have removed the goto and continue statements.

Reply:
fixed